internal/cluster/cluster.go

```diff
@@ -668,6 +674,70 @@ func (m *Manager) UpdateLogsProgress(ctx context.Context, c *common.Cluster, pro
 	return err
 }
 
+func (m *Manager) UpdateInstallProgress(ctx context.Context, clusterID strfmt.UUID) error {
+	log := logutil.FromContext(ctx, m.log)
+
+	cluster, err := common.GetClusterFromDB(m.db, clusterID, common.UseEagerLoading)
+	if err != nil {
+		log.WithError(err).Error("Failed to get cluster from DB")
+		return err
+	}
+
+	var totalHostsDoneStages, totalHostsStages float64
+	for _, h := range cluster.Hosts {
+		stages := m.hostAPI.GetStagesByRole(h.Role, h.Bootstrap)
+		currentIndex := m.hostAPI.IndexOfStage(h.Progress.CurrentStage, stages)
```

Comment:
Is -1 a valid index here?

Reply:
Yes, as you can see I add `+1`, that means that the first stage gets 1 instead of 0 and therefore we get a real positive percentage.

If the index doesn't exist, we will get `-1 + 1 = 0` therefore the progress will be `0`.